cnxdb/contrib/pytest.py

```diff
@@ -87,6 +87,12 @@ def db_init(db_engines):
     """Initializes the database"""
     from cnxdb.init.main import init_db
     init_db(db_engines['super'], is_venv_importable())
+    if is_venv_importable():
+        # We need to recreate the connection pool after initializing venv
+        # because all the existing connections wouldn't have venv set up
+        for key in db_engines:
+            db_engines[key].pool.dispose()
+            db_engines[key].pool.recreate()
```

Comment:
I've done something similar to this in my recent branch. Though, I wonder why you choose to use `.pool.dispose` and `.pool.recreate` rather than just use `.dispose` on the engine itself? From what I've seen, the `dispose` method does the trick.

Reply:
From what I see of the sqlalchemy api, either `<engine>.dispose()` or what you've done here have the same effect. Note that it only works on connections that are not active, in both cases. Makes me worry about stuff like this:

Reply:
I think it's because `db_engine[key].dispose()` completely gets rid of the engine so I had to store the url and then use `create_engine` again.  I don't want to do that in case we pass in more arguments to `create_engine` in the future.  I don't know if it's because I'm not doing it correctly.